### DISCUSSION TITLE
Indexing and thread safety when using workers

### DISCUSSION THREAD

**Discussion by letmaik** [microsoft/CCF#4037] [General]
https://microsoft.github.io/CCF/main/architecture/threading.html#programming-model says a bit about the programming model when using the experimental `worker_threads` option.

When using an [indexer](https://microsoft.github.io/CCF/main/build_apps/api.html#indexing) then committed transactions are processed in the background and the state of the indexer is continuously updated.

With multiple worker threads, do indexers have to be made thread-safe or are indexers only being run between rpc calls to avoid this issue altogether?

**Comment by achamayou** [comment]
@letmaik the state that the indexers expose to historical queries does need to be made thread-safe.

**Reply by letmaik** [reply]
Do the built-in indexers expose state thread-safe?

**Reply by achamayou** [reply]
They do not, filed as #4038.

**Comment by achamayou** [comment]
Closed by #4075.
